packages/cli/src/upgrade/upgrade.js

```diff
@@ -4,151 +4,215 @@
  * LICENSE file in the root directory of this source tree.
  *
- * @format
  * @flow
  */
 
-import fs from 'fs';
+import https from 'https';
 import path from 'path';
+import fs from 'fs';
 import semver from 'semver';
+import execa from 'execa';
 import type { ContextT } from '../core/types.flow';
 import logger from '../util/logger';
-import copyProjectTemplateAndReplace from '../generator/copyProjectTemplateAndReplace';
+import PackageManager from '../util/PackageManager';
+
+const fetch = (url: string) =>
+  new Promise((resolve, reject) => {
+    const request = https.get(url, response => {
+      if (response.statusCode < 200 || response.statusCode > 299) {
+        reject(
+          new Error(`Failed to load page, status code: ${response.statusCode}`)
+        );
+      }
+      const body = [];
+      response.on('data', chunk => body.push(chunk));
+      response.on('end', () => resolve(body.join('')));
+    });
+    request.on('error', err => reject(err));
+  });
+
+const getLatestRNVersion = async (): Promise<string> => {
+  logger.info('No version passed. Fetching latest...');
+  const { stdout } = await execa('npm', ['info', 'react-native', 'version']);
+  return stdout;
+};
 
-/**
- * Migrate application to a new version of React Native.
- * See http://facebook.github.io/react-native/docs/upgrading.html
- */
-function validateAndUpgrade(argv: Array<string>, ctx: ContextT) {
-  const projectDir = ctx.root;
+const getRNPeerDeps = async (
+  version: string
+): Promise<{ [key: string]: string }> => {
+  const { stdout } = await execa('npm', [
+    'info',
+    `react-native@${version}`,
+    'peerDependencies',
+    '--json',
+  ]);
+
+  return JSON.parse(stdout);
+};
 
-  const packageJSON = JSON.parse(
-    fs.readFileSync(path.resolve(projectDir, 'package.json'), 'utf8')
-  );
+const getPatch = async (currentVersion, newVersion, projectDir) => {
+  let patch;
+  const rnDiffPurgeUrl = 'https://github.com/pvinis/rn-diff-purge';
+  const rnDiffAppName = 'RnDiffApp';
+  const { name } = require(path.join(projectDir, 'package.json'));
 
-  warn(
-    'You should consider using the new upgrade tool based on Git. It ' +
-      'makes upgrades easier by resolving most conflicts automatically.\n' +
-      'To use it:\n' +
-      '- Go back to the old version of React Native\n' +
-      '- Run "npm install -g react-native-git-upgrade"\n' +
-      '- Run "react-native-git-upgrade"\n' +
-      'See https://facebook.github.io/react-native/docs/upgrading.html'
-  );
+  logger.info(`Fetching diff between v${currentVersion} and v${newVersion}...`);
 
-  const projectName = packageJSON.name;
-  if (!projectName) {
-    warn(
-      'Your project needs to have a name, declared in package.json, ' +
-        'such as "name": "AwesomeApp". Please add a project name. Aborting.'
+  try {
+    patch = await fetch(
+      `${rnDiffPurgeUrl}/compare/version/${currentVersion}...version/${newVersion}.diff`
```

Comment:
This actually works for jumping multiple versions?

Reply:
Yep, see the screenshots

Reply:
Yes, it's just a diff between more commits. Works fine.